api/src/metadata_json.rs

```diff
@@ -14,22 +14,54 @@ use crate::{
 #[derive(Clone, Debug)]
 pub struct MetadataJson {
     metadata_json: MetadataJsonInput,
-    collection: Uuid,
     uri: Option<String>,
     identifier: Option<String>,
 }
 
 impl MetadataJson {
     #[must_use]
-    pub fn new(collection: Uuid, metadata_json: MetadataJsonInput) -> Self {
+    pub fn new(metadata_json: MetadataJsonInput) -> Self {
         Self {
             metadata_json,
-            collection,
             uri: None,
             identifier: None,
         }
     }
 
+    /// Fetches metadata from the database and constructs a `Self` instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `id` - The ID of the metadata to fetch.
+    /// * `db` - The database connection to use.
+    ///
+    /// # Errors
+    ///
+    /// This function fails if there is no matching `metadata_json` entry in the database
+    /// or if it is unable to fetch related data from the database
+    pub async fn fetch(id: Uuid, db: &Connection) -> Result<Self> {
```

Comment:
Nice. Good job on docs too. We are going to publish cargo docs soon for our services.